src/Nox.Generator/Domain/EntityMetaGenerator.cs

```diff
@@ -21,63 +25,74 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
         foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
         {
             context.CancellationToken.ThrowIfCancellationRequested();
-
-            var propertiesWithSource = entity.GetAllMembers()
-                .Select( t => new { TypeDef = t.Value, Source = GenerateStaticTypeOptions(t.Value, codeGeneratorState.Solution) })
-                .Where( t => t.Source != null)
+            
+            var typeDef = entity.GetAllMembers()
+                .Select( t =>  GenerateStaticTypeOptions(t.Value, codeGeneratorState.Solution) )
                 .ToList();
-
+            
             new TemplateCodeBuilder(context, codeGeneratorState)
                 .WithClassName(entity.Name)
                 .WithFileNamePrefix($"Domain.Meta")
                 .WithObject("entity", entity)
-                .WithObject("propertiesWithSource", propertiesWithSource)
+                .WithObject("typeDef", typeDef)
                 .GenerateSourceCodeFromResource("Domain.EntityMeta");
         }
     }
 
-    private string? GenerateStaticTypeOptions(NoxSimpleTypeDefinition typeDef, NoxSolution solution)
+    private EntityMetaData GenerateStaticTypeOptions(NoxSimpleTypeDefinition typeDef, NoxSolution solution)
     {
-
         var type = typeDef.Type == NoxType.EntityId ? solution.GetSingleKeyTypeForEntity(typeDef.EntityIdTypeOptions!.Entity) : typeDef.Type;
 
         var typeOptions = $"{type}TypeOptions";
 
         var options = typeof(NoxSimpleTypeDefinition).GetProperty(typeOptions);
 
         var optionsOutput = string.Empty;
+        
+        var optionsProperties = Array.Empty<string>();
 
         var components = type.GetComponents(typeDef).ToArray();
 
         var inParams = components.Length == 1 
             ? $"{components[0].Value.FullName} value" 
             : $"I{type} value" ;
 
-        string? factoryOutput;
-
+       
         if (options != null)
         {
-            var optionsValue = options.GetValue(typeDef, null);
-
-            if (optionsValue != null)
-            {
-                optionsOutput = "public static " +
-                    optionsValue.ToSourceCode($"{typeDef.Name}TypeOptions {{get; private set;}}") + "\r\n";
+            optionsProperties = GetOptionsOutputAndProperties(typeDef, options, optionsProperties, ref optionsOutput);
+        }
+        
+        return new EntityMetaData
+        {
+            Name = typeDef.Name,
+            Type = type.ToString(),
+            InParams = inParams,
+            OptionsOutput = optionsOutput,
+            OptionsProperties = optionsProperties
+        };
+    }
 
-                factoryOutput =
-                    $"public static {type} Create{typeDef.Name}({inParams})\r\n" +
-                    $"    => Nox.Types.{type}.From(value, {typeDef.Name}TypeOptions);" +
-                    $"\r\n\r\n";
+    private static string[] GetOptionsOutputAndProperties(
```

Comment:
the refactor of the template it to avoid this kind of code and be explicitly on the template what code we are generating